### PR TITLE
Various updates.

### PR DESCRIPTION
**Pull request by sfshaza2**
Fixed miscellaneous things, namely:

- spellings/typos from PR https://github.com/flutter/website/pull/4943
- updated the news2 image/link on the landing page, as per marketing
- fixed various URL links to use our macros.
- added a new macro for the apple developer site.
- also fixed a jekyll error in /docs/perf/metrics.md

### CODE REVIEW

File: src/desktop.md [hunk 1]
```diff
@@ -494,7 +494,7 @@ see the following resources:
 [Developing packages and plugins]: /docs/development/packages-and-plugins/developing-packages
 [Federated Plugin proposal]: /go/federated-plugins
 [Federated plugins]: /docs/development/packages-and-plugins/developing-packages#federated-plugins
-[How to write a Flutter web plugin, part 2]: https://medium.com/flutter/how-to-write-a-flutter-web-plugin-part-2-afdddb69ece6
+[How to write a Flutter web plugin, part 2]: {{site.medium}}/flutter/how-to-write-a-flutter-web-plugin-part-2-afdddb69ece6
```
Comment (kwalrath) [inline src/desktop.md:497]: It looks like there's a separate medium-flutter macro that we could/should be using here:

```suggestion
[How to write a Flutter web plugin, part 2]: {{site.medium-flutter}}/how-to-write-a-flutter-web-plugin-part-2-afdddb69ece6
```

File: src/docs/development/add-to-app/android/add-flutter-view.md [hunk 2]
```diff
@@ -33,46 +33,46 @@ class="mw-100" alt="Add Flutter View sample video" %}
...
   is visible.
 
-The reverse [detachFromFlutterEngine](https://api.flutter.dev/javadoc/io/flutter/embedding/android/FlutterView.html#detachFromFlutterEngine--) and other lifecycle methods on the [LifecycleChannel](https://api.flutter.dev/javadoc/io/flutter/embedding/engine/systemchannels/LifecycleChannel.html)
+The reverse [detachFromFlutterEngine]({{site.api}}/javadoc/io/flutter/embedding/android/FlutterView.html#detachFromFlutterEngine--) and other lifecycle methods on the [LifecycleChannel]({{site.api}}/javadoc/io/flutter/embedding/engine/systemchannels/LifecycleChannel.html)
 class must also be called to not leak resources when the FlutterView or Activity
 is no longer visible.
 
-In addition, see the remaining implementation in the [FlutterViewEngine](https://github.com/flutter/samples/blob/master/add_to_app/android_view/app/src/main/java/dev/flutter/example/androidView/FlutterViewEngine.kt)
+In addition, see the remaining implementation in the [FlutterViewEngine]({{site.api}}/flutter/samples/blob/master/add_to_app/android_view/app/src/main/java/dev/flutter/example/androidView/FlutterViewEngine.kt)
```
Comment (kwalrath) [inline src/docs/development/add-to-app/android/add-flutter-view.md:75]:
```suggestion
In addition, see the remaining implementation in the [FlutterViewEngine]({{site.github}}/flutter/samples/blob/master/add_to_app/android_view/app/src/main/java/dev/flutter/example/androidView/FlutterViewEngine.kt)
```

File: src/docs/development/platform-integration/platform-channels.md [hunk 3]
```diff
@@ -801,9 +801,9 @@ DispatchQueue.main.async {
...
 [`StringCodec`]: {{site.api}}/flutter/services/StringCodec-class.html
-[the main thread]: https://developer.apple.com/documentation/uikit?language=objc
-[the UI thread]: https://developer.android.com/guide/components/processes-and-threads#Threads
+[the main thread]: {{site.apple-dev}}/documentation/uikit?language=objc
+[the UI thread]: {{site.android-dev}}/guide/components/processes-and-threads#Threads
 [using packages]: /docs/development/packages-and-plugins/using-packages
-[Pigeon]: https://pub.dev/packages/pigeon
-[Pigeon pub.dev page]: https://pub.dev/packages/pigeon
+[Pigeon]: {{site.pub}}/packages/pigeon
+[Pigeon pub.dev page]: {{site.pub}}/packages/pigeon
```
Comment (kwalrath) [inline src/docs/development/platform-integration/platform-channels.md:808]: Could also combine these...

```suggestion
[Pigeon]: {{site.pub-pkg}}/pigeon
[Pigeon pub.dev page]: {{site.pub-pkg}}/pigeon
```

File: src/docs/development/tools/hot-reload.md [hunk 4]
```diff
@@ -360,7 +360,7 @@ widgets and render objects.
 
 
 [const-new]: https://news.dartlang.org/2012/06/const-static-final-oh-my.html
-[Dart Virtual Machine (VM)]: https://dart.dev/platforms
+[Dart Virtual Machine (VM)]: {{site.dart-site}}/platforms
```
Comment (kwalrath) [inline src/docs/development/tools/hot-reload.md:363]: We've removed this page, fyi... It'll get redirected to https://dart.dev/overview#platform.

File: src/docs/perf/faq.md [hunk 5]
```diff
@@ -3,82 +3,84 @@ title: Frequently asked questions
...
+  * [How to write a memory test for Flutter]({{site.github}}/flutter/flutter/wiki/How-to-write-a-memory-test-for-Flutter)
+
+* What are some tools for capturing and analyzing performance metrics?
+  * [Dart DevTools](/docs/development/tools/devtools)
+  * [Apple instruments](https://en.wikipedia.org/wiki/Instruments_(software))
+  * [Linux perf](https://en.wikipedia.org/wiki/Perf_(Linux))
+  * [Chrome tracing (enter `about:tracing` in your Chrome URL field)][tracing]
+  * [Android systrace (`adb systrace`)][systrace]
+  * [Fuchsia `fx traceutil`][traceutil]
+  * [Dart observatory](https://dart-lang.github.io/observatory/)
```
Comment (kwalrath) [inline src/docs/perf/faq.md:23]: This site was just deleted.

```suggestion
```

File: src/docs/perf/faq.md [hunk 6]
```diff
@@ -3,82 +3,84 @@ title: Frequently asked questions
...
-- How can I take a screenshot of an app that is running and export it as a SKP
-  file?
-  - Run `flutter screenshot --type=skia --observotry-uri=...`
-  - Known issue of a SKP screenshot:
-    - Doesn’t record images in real devices [#21237](https://github.com/flutter/flutter/issues/21237).
-  - To analyze and visualize the SKP file, visit https://debugger.skia.org/.
-
-- How do I retrieve the shader persistent cache from a device?
-  - On Android, you can do the following:
+    [GitHub commits]({{site.github}}/flutter/engine/commits).
```
Comment (kwalrath) [inline src/docs/perf/faq.md:62]:
```suggestion
    [flutter/engine GitHub commits]({{site.github}}/flutter/engine/commits).
```

File: src/docs/perf/faq.md [hunk 7]
```diff
@@ -3,82 +3,84 @@ title: Frequently asked questions
...
+* How do I determine my Flutter app’s package size that will be
+  downloaded by a user?
+  * See [Measuring your app's size](/docs/perf/app-size)
 
-- How do I see the breakdown of the Flutter engine size?
-  - Visit
-    https://storage.googleapis.com/flutter_infra/flutter/c3976b3c7183f479717bffed3f640fb92afbd3dc/android-arm-release/sizes/index.html,
+* How do I see the breakdown of the Flutter engine size?
+  * Visit the
+    [Binary size dashboard](https://storage.googleapis.com/flutter_infra/flutter/c3976b3c7183f479717bffed3f640fb92afbd3dc/android-arm-release/sizes/index.html),
```
Comment (kwalrath) [inline src/docs/perf/faq.md:60]:
```suggestion
    [binary size dashboard](https://storage.googleapis.com/flutter_infra/flutter/c3976b3c7183f479717bffed3f640fb92afbd3dc/android-arm-release/sizes/index.html),
```

File: src/docs/perf/metrics.md [hunk 8]
```diff
@@ -1,80 +1,71 @@
...
+  * See the [basic_material_app_android][], [basic_material_app_ios][],
     [hello_world_android][], [hello_world_ios][], [flutter_gallery_android][],
-    [flutter_gallery_ios][] tests.
-  - See [metrics][size_perf] in the dashboard.
-  - For how to measure the size more accurately, see 
-    https://flutter.dev/docs/perf/app-size.
+    and [flutter_gallery_ios][] tests.
+  * See [metrics][size_perf] in the dashboard.
+  * For info on how to measure the size more accurately,
+    see the page on [app-size](/docs/perf/app-size).
```
Comment (kwalrath) [inline src/docs/perf/metrics.md:40]:
```suggestion
    see the [app size](/docs/perf/app-size) page.
```

File: src/index.html [hunk 9]
```diff
@@ -287,8 +287,8 @@ <h3>Announcing Flutter 1.22</h3>
         {% asset homepage/news-2.png alt='A news item image' class="card-img-top" %}
         <div class="card-body">
             <p class="card-text"><small class="text-muted">News</small></p>
-            <h3>Announcing Flutter Windows Alpha</h3>
-            <a href="https://medium.com/flutter/announcing-flutter-windows-alpha-33982cd0f433">Read more</a>
+            <h3>Sign up for Flutter Engage on 3/3!</h3>
```
Comment (kwalrath) [inline src/index.html:290]: Don't want to imply that the signup is 3/3...

```suggestion
            <h3>Sign up for Flutter Engage, happening 3/3!</h3>
```

File: src/web.md [hunk 10]
```diff
@@ -15,10 +15,11 @@ and you don’t need a browser plug-in.
   If you experience a problem that hasn't yet been reported, please
   [file an issue][] and make sure that "web" appears in the title.
 {{site.alert.end}}
+
 {{site.alert.warning}}
   **Hot reload not still supported**
-  At the current beta state, Flutter web only support **hot restart**,
-  no support for **hot reload** is available yet.
+  At of beta, web supports only **hot restart**;
```
Comment (kwalrath) [inline src/web.md:21]:
```suggestion
  As of beta, only **hot restart** is supported on the web
```

File: src/web.md [hunk 11]
```diff
@@ -15,10 +15,11 @@ and you don’t need a browser plug-in.
   If you experience a problem that hasn't yet been reported, please
   [file an issue][] and make sure that "web" appears in the title.
 {{site.alert.end}}
+
 {{site.alert.warning}}
   **Hot reload not still supported**
```
Comment (kwalrath) [inline src/web.md:20]:
```suggestion
  **Hot reload not supported**
```

**Review by kwalrath — APPROVED** [review]
Some minor issues; otherwise lgtm

**Comment by kwalrath** [comment]
Note that I didn't try rendering these pags. I hope you did. :)

File: src/docs/perf/faq.md (hunk 5 above)
Comment (sfshaza2) [reply src/docs/perf/faq.md:23]: Ack. I definitely didn't approve that.

File: src/web.md [hunk 12]
```diff
@@ -15,10 +15,11 @@ and you don’t need a browser plug-in.
   If you experience a problem that hasn't yet been reported, please
   [file an issue][] and make sure that "web" appears in the title.
 {{site.alert.end}}
+
 {{site.alert.warning}}
-  **Hot reload not still supported**
-  At the current beta state, Flutter web only support **hot restart**,
-  no support for **hot reload** is available yet.
+  **Hot reload not supported**
+  As of beta, only **hot restart** is supported on the web
```
Comment (kwalrath) [inline src/web.md:21]:
```suggestion
  As of beta, only **hot restart** is supported on the web;
```

File: src/web.md [hunk 13]
```diff
@@ -15,10 +15,11 @@ and you don’t need a browser plug-in.
   If you experience a problem that hasn't yet been reported, please
   [file an issue][] and make sure that "web" appears in the title.
 {{site.alert.end}}
+
 {{site.alert.warning}}
-  **Hot reload not still supported**
-  At the current beta state, Flutter web only support **hot restart**,
-  no support for **hot reload** is available yet.
+  **Hot reload not supported**
```
Comment (kwalrath) [inline src/web.md:20]:
```suggestion
  **Hot reload not supported:**
```

**Review by kwalrath — APPROVED** [review]
lgtm, but I think one note needs a couple of tweaks

File: src/desktop.md [hunk 14]
```diff
@@ -516,7 +516,7 @@ see the following resources:
 [Developing packages and plugins]: /docs/development/packages-and-plugins/developing-packages
 [Federated Plugin proposal]: /go/federated-plugins
 [Federated plugins]: /docs/development/packages-and-plugins/developing-packages#federated-plugins
-[How to write a Flutter web plugin, part 2]: https://medium.com/flutter/how-to-write-a-flutter-web-plugin-part-2-afdddb69ece6
+[How to write a Flutter web plugin, part 2]: {{site.medium-flutter}}/how-to-write-a-flutter-web-plugin-part-2-afdddb69ece6
```
Comment (kwalrath) [inline src/desktop.md:519]:
```suggestion
[How to write a Flutter web plugin, part 2]: {{site.flutter-medium}}/how-to-write-a-flutter-web-plugin-part-2-afdddb69ece6
```

File: src/docs/development/data-and-backend/json.md [hunk 15]
```diff
@@ -529,7 +529,7 @@ For more information, see the following resources:
 [`dart:convert`]: {{site.dart.api}}/{{site.dart.sdk.channel}}/dart-convert
 [`explicitToJson`]: {{site.pub}}/documentation/json_annotation/latest/json_annotation/JsonSerializable/explicitToJson.html
 [Flutter Favorite]: /docs/development/packages-and-plugins/favorites
-[json background parsing]: https://flutter.dev/docs/cookbook/networking/background-parsing
+[json background parsing]: /dev/docs/cookbook/networking/background-parsing
```
Comment (kwalrath) [inline src/docs/development/data-and-backend/json.md:532]:
```suggestion
[json background parsing]: /docs/cookbook/networking/background-parsing
```

**Pull request merged**